GitUI/UserControls/BlameControl.cs

```diff
@@ -457,33 +457,50 @@ private void copyCommitHashToClipboardToolStripMenuItem_Click(object sender, Eve
             CopyToClipboard(c => c.ObjectId.ToString());
         }
 
-        private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventArgs e)
+        private bool TryGetSelectedRevision(out GitRevision selectedRevision)
         {
+            selectedRevision = null;
             int line = (int?)contextMenu.Tag ?? -1;
             if (line < 0)
             {
-                return;
+                return false;
             }
 
             var objectId = _blame.Lines[line].Commit.ObjectId;
-            int originalLine = _blame.Lines[line].OriginLineNumber;
-            GitBlame blame = Module.Blame(_fileName, objectId + "^", _encoding, originalLine + ",+1");
-            if (blame.Lines.Count > 0)
+
+            selectedRevision = _revGrid.GetRevision(objectId);
+            return selectedRevision != null;
+        }
+
+        private void blameRevisionToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (!TryGetSelectedRevision(out var selectedRevision))
             {
-                var revision = blame.Lines[0].Commit.ObjectId;
-                if (_revGrid != null)
-                {
-                    _clickedBlameLine = blame.Lines[0];
-                    _revGrid.SetSelectedRevision(revision);
-                }
-                else
+                return;
+            }
+
+            _revGrid.SetSelectedRevision(selectedRevision);
+        }
+
+        private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            if (!TryGetSelectedRevision(out var selectedRevision))
+            {
+                return;
+            }
+
+            if (!selectedRevision.HasParent)
+            {
+                _revGrid.SetSelectedRevision(selectedRevision);
+                using (var frm = new FormCommitDiff(UICommands, selectedRevision.ObjectId))
                 {
-                    using (var frm = new FormCommitDiff(UICommands, revision))
-                    {
-                        frm.ShowDialog(this);
-                    }
+                    frm.ShowDialog(this);
```

Comment:
Before this change, `FormCommitDiff` was opened if `_revGrid == null`.

It should not be done if there is no parent commit. Instead consider opening it if `_revGrid == null` in _both_ `blameRevisionToolStripMenuItem_Click` and `blamePreviousRevisionToolStripMenuItem_Click`.